src/components/token_row_filter.tsx

```diff
@@ -0,0 +1,124 @@
+import _ from "lodash";
+import React from "react";
+
+import { ColorOption } from "../style/theme";
+import { TokenInfo } from "../types";
+import { Circle } from '../components/ui/circle';
+import { Container } from '../components/ui/container';
+import { Flex } from '../components/ui/flex';
+import { Text } from '../components/ui/text';
+import { tokenUtils } from "../util/token";
+
+interface TokenRowFilterProps {
+    tokens: TokenInfo[];
+    onClick: (token: TokenInfo) => void;
+    searchQuery: string;
+}
+
+export class TokenRowFilter extends React.Component<TokenRowFilterProps> {
+    public render(): React.ReactNode {
+        return _.map(this.props.tokens, token => {
+            if (!this._isTokenQueryMatch(token)) {
+                return null;
+            }
+            return <TokenSelectorRow key={token.address} token={token} onClick={this.props.onClick} />;
+        });
+    }
+    public shouldComponentUpdate(nextProps: TokenRowFilterProps): boolean {
+        const arePropsDeeplyEqual = _.isEqual(nextProps, this.props);
```

Comment:
Why is this expensive operation needed? If you need to prevent references from changing unnecessarily you can use `useCallback` & `useMemo`

Reply:
There is still a mix between the old class code style and the new hook based. I will migrate this portion to the hook based